### PR TITLE
[CI] github workflow to test with specific istio version

### PR DESCRIPTION
**Pull request by jmazzitelli**
This provides a new workflow that can be triggered manually. It will allow you to run the integration tests with any version of Istio you want (including even dev builds of Istio that have not been released, such as `1.20-dev`).

This will provide us the ability to see if Kiali can support older versions (or newer versions in development) of Istio.

You can test master branch or any other branch you select.

### CODE REVIEW

**Review by leandroberetta — DISMISSED** [review]
Nice addition! Looks good to me.

**Comment by jmazzitelli** [comment]
I will merge once this PR goes green as well as my fork test (which is running a test with istio 1.20-dev build): https://github.com/jmazzitelli/kiali/actions/runs/6262913992

**Comment by jmazzitelli** [comment]
I found a spot I missed. Will re-test on my fork.  Will wait for this PR to go green AND the fork testing to be successful before merging.

**Comment by jmazzitelli** [comment]
Everything passes except for some odd flakes that do not look related. the proper istio version successfully started and most passes (backend and frontend). I'm seeing things like
```
1) Kiali App Details page
...
AssertionError: Timed out retrying after 40000ms: Expected <svg#loading_kiali_spinner.pf-v5-c-spinner.pf-m-lg> not to exist in the DOM, but it was continuously found.
```

I'm going to assume this PR is good and this is a flake or our tests do not yet pass on 1.20 (which is exactly what this PR is meant to flush out).

**Review by leandroberetta — APPROVED** [review]

**Pull request merged**
